docs/api-usage.rst

```diff
@@ -423,6 +423,7 @@ python-gitlab can automatically retry in such case, when
 HTTP error codes 500 (Internal Server Error), 502 (502 Bad Gateway),
 503 (Service Unavailable), and 504 (Gateway Timeout) are retried.  By
 default an exception is raised for these errors.
+It will retry until reaching `max_retries` value.
```

Comment:
Thanks fa lot or this @emirot! I think it might be a bit confusing if we state the retry behavior after we mention the default (no retry) behavior - maybe your PR is a chance to reword this a bit. Feel free to also shuffle it around and put the default behavior wording higher up if you feel it's more appropriate.

```suggestion
503 (Service Unavailable), and 504 (Gateway Timeout) are retried. It will retry until reaching
the `max_retries` value. By default, `retry_transient_errors` is set to `False` and an exception
is raised for these errors.
```